x-pack/docs/en/security/authentication/remote-clusters-privileges.asciidoc

```diff
@@ -0,0 +1,295 @@
+<<remote-clusters-connect,connecting to the remote cluster>> using
+the cluster update settings API, you will also need a user with the `all`
+cluster privilege.
+
+The following request creates a `remote-replication` role on the local cluster:
+
+[source,console]
+----
+POST /_security/role/remote-replication
+{
+  "cluster": [
+    "manage_ccr"
+  ],
+  "indices": [
+    {
+      "names": [
+        "follower-index-name"
+      ],
+      "privileges": [
+        "monitor",
+        "read",
+        "write",
+        "manage_follow_index"
+      ]
+    }
+  ]
+}
+----
+
+After creating the `remote-replication` role on each cluster, use the 
+<<security-api-put-user,create or update users API>> to create a user on
+each cluster and assign the `remote-replication` role. For example, the
+following request assigns the `remote-replication` role to a user named
+`cross-cluster-user`:
+
+[source,console]
+----
+POST /_security/user/cross-cluster-user
+{
+  "password" : "l0ng-r4nd0m-p@ssw0rd",
+  "roles" : [ "remote-replication" ]
+}
+----
+// TEST[continued]
+
+//end::configure-ccr-privileges[]
+
+You can then <<ccr-getting-started-tutorial,configure {ccr}>> to replicate your
+data across datacenters.
+
+[[remote-clusters-privileges-ccs]]
+==== Configure privileges for {ccs}
+The {ccs} user requires different cluster and index privileges on the remote
+cluster and local cluster. The following requests create separate roles on the
+local and remote clusters, and then create a user with the required roles.
+
+[discrete]
+===== Remote cluster
+On the remote cluster, the {ccs} role requires the `read` and
+`read_cross_cluster` privileges for the target indices.
+
+NOTE: If requests will be issued <<run-as-privilege,on behalf of other users>>,
+then the the authenticating user must have the `run_as` privilege on the remote 
+cluster.
+
+The following request creates a `remote-search` role on the remote cluster:
+
+[source,console]
+----
+POST /_security/role/remote-search
+{
+  "indices": [
+    {
+      "names": [
+        "target-indices"
+      ],
+      "privileges": [
+        "read",
+        "read_cross_cluster"
+      ]
+    }
+  ]
+}
+----
+
+////
+[source,console]
+----
+DELETE /_security/role/remote-search
+----
+// TEST[continued]
+////
+
+[discrete]
+===== Local cluster
+On the local cluster, which is the cluster used to initiate cross cluster
+search, a user only needs the `remote-search` role. The role privileges can be
+empty.
+
+The following request creates a `remote-search` role on the remote cluster:
+
+
+[source,console]
+----
+POST /_security/role/remote-search
+{}
+----
+
+After creating the `remote-replication` role on each cluster, create a user on
+each cluster and assign the `remote-replication` role. 
+
+After creating the `remote-search` role on each cluster, use the 
+<<security-api-put-user,create or update users API>> to create a user on each 
+cluster and assign the `remote-search` role. For example, the following request 
```

Comment:
Technically this user is not needed on the remote cluster. Also the `remote-search` does not need to be defined on the local cluster either. But I am not sure whether explain all these would end up confusing users instead of being helpful. So maybe it's OK to leave it as is.

Reply:
@ywangd, so the `remote-search` role and the `cross-search-user` are only required on the remote clusters? This differs from CCR then, since the role and user are only required on the remote clusters. Is it harmful or potentially error-inducing to have that role and user on the local cluster as well?

Reply:
The `role` is required on the remote cluster, but the user is required on the remote cluster. I didn't test but I believe for CCR, the user is also only required on the local cluster. The difference is that the role needs to be defined on both local and remote clusters. I don't think it's harmful or really error prone. My only (minor) concern is that users may find out, say, ccs-user is not needed on the remote cluster and either suspect the doc is accurate or the software is behaving (both would put us in a situation that may require more explanation or even SDH, we had similar one for read-cross-cluster privilege).

Reply:
I tested this setup, and it seems that you're correct! The role (with different privileges) is required on both the local and remote clusters, but creating the user is only required on the local cluster. The docs should reflect how the feature actually works, so I'll modify this information and also add a note that you only need to create this user on the *local* cluster.